submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/helpers/SourceGenerator.kt

```diff
@@ -59,16 +63,18 @@ class SourceGenerator(
 
             if (sub != null) {
                 add(fireSourceFactory.createSubmissionFireSource(sub.accNo, Paths.get("${sub.relPath}/Files")))
-                add(PathFilesSource(Paths.get(props.submissionPath).resolve(sub.relPath).resolve(FILES_PATH)))
+
+                val subPath = Paths.get(props.submissionPath).resolve("${sub.relPath}/$FILES_PATH")
+                add(PathSource("Submission ${sub.accNo} previous version files", subPath))
```

Comment:
Since we already have the submission and thinking that we'll support history, I think it might add value to include the version as well so it's more specific

`Files for submission ${sub.accNo} version ${sub.version}`

Reply:
This error is for users and they are not aware of the concept "version". That is the reason I decide to keep in generic.